src/Streamly/Internal/Data/Stream/StreamD.hs

```diff
@@ -1528,15 +1528,129 @@ groupsBy :: Monad m
     -> Fold m a b
     -> Stream m a
     -> Stream m b
-groupsBy cmp f = foldMany (FL.groupBy cmp f)
+groupsBy cmp f (Stream step state) = Stream (stepOuter f) (Just state, Nothing)
```

Comment:
We can define this as:

```
groupsBy cmp f = parseMany (PR.groupBy cmp f)
```

it can accept a Parser instead of a Fold. Anyway folds can be upgraded to parsers. We only need to make sure that the performance is not affected.

Reply:
Yeah, once they are added we can change it.

Reply:
We will do it in a future change. The parser functions are still not defined I believe.